### PR TITLE
Record a possibly incomplete batch

### PR DESCRIPTION
**Pull request by WilliamBZA**

### CODE REVIEW

File: src/ServiceControl/Recoverability/Grouping/Retries/RetryOperationManager.cs [hunk 1]
```diff
@@ -83,6 +95,7 @@ public class RetryOperationSummary
     {
         public int? MessagesRemaining { get; internal set; }
 
+        public bool IsPossiblyIncomplete { get; set; } // Need a better name for this
```
Comment (janovesk) [inline src/ServiceControl/Recoverability/Grouping/Retries/RetryOperationManager.cs]: How about `WasInterrupted`?

File: src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs [hunk 2]
```diff
@@ -84,7 +84,8 @@ dynamic GetAllGroups(string classifier)
                             First = failureGroup.First,
                             Last = failureGroup.Last,
                             RetryStatus = summary != null ? "In Progress" : null,
-                            NumberOfRetryMessagesRemaining = summary?.MessagesRemaining
+                            NumberOfRetryMessagesRemaining = summary?.MessagesRemaining,
+                            IsPossiblyAnIncompleteRetry = summary.IsPossiblyIncomplete
```
Comment (janovesk) [inline src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs]: Don't we know for a fact that it WAS interrupted?

File: src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs [hunk 3]
```diff
@@ -114,26 +114,33 @@ internal void AdoptOrphanedBatches(IDocumentSession session, out bool hasMoreWor
         {
             RavenQueryStatistics stats;
 
-            var orphanedBatchIds = session.Query<RetryBatch, RetryBatches_ByStatusAndSession>()
+            var orphanedBatches = session.Query<RetryBatch, RetryBatches_ByStatusAndSession>()
```
Comment (janovesk) [inline src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs]: Not sure how smart Raven is for this. Will this change what's going over the wire from a list of id's to a list of full documents?

File: src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs [hunk 4]
```diff
@@ -114,26 +114,33 @@ internal void AdoptOrphanedBatches(IDocumentSession session, out bool hasMoreWor
...
 
-            log.InfoFormat("Found {0} orphaned retry batches from previous sessions", orphanedBatchIds.Length);
+            log.InfoFormat("Found {0} orphaned retry batches from previous sessions", orphanedBatches.Length);
 
-            AdoptBatches(session, orphanedBatchIds);
+            AdoptBatches(session, orphanedBatches.Select(b => b.Id));
+
+            foreach (var batch in orphanedBatches)
+            {
+                if (batch.RetryType != RetryType.MultipleMessages)
```
Comment (janovesk) [inline src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs]: I'd drop this if check if that means we have to load the full document compared to just an Id previously. No harm in having status for `RetryType.MultipleMessages` in the RetryOperationManager in-memory cache. They won't be queried, but will be cleared out when the adopted batch is forwarded.

File: src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs (hunk 2 above)
Comment (WilliamBZA) [inline src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs]: No, all the documents could have been marked and SC could have crashed before updating the batch status

File: src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs (hunk 4 above)
Comment (WilliamBZA) [inline src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs]: We still need to load the retry type and the request id to mark the correct batch

File: src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs (hunk 2 above)
Comment (janovesk) [inline src/ServiceControl/Recoverability/Grouping/FailureGroupsApi.cs]: So you're defining updating of the batch status as not part of the retry? :D

**Pull request merged**